src/lib.rs

```diff
@@ -425,6 +431,14 @@ pub mod common {
         }
     }
 
+    impl<S: Index + Len> IntoIterator for Slice<S> {
+        type Item = S::Ref;
+        type IntoIter = IterOwn<Slice<S>>;
+        fn into_iter(self) -> Self::IntoIter {
+            IterOwn { index: 0, slice: self }
```

Comment:
```suggestion
            self.into_index_iter()
```

Reply:
Folded in to a commit with the other feedback; tyvm!